queue_tracker_test.go

```diff
@@ -34,21 +31,23 @@ func TestQueueTracker(t *testing.T) {
 	}
 
 	allJobRegistered := false
+	firstJobInsertNotify := make(chan struct{})
```

Comment:
You can define this variable simply as `make(chan bool)`, so you can send into channel whether `q2` has registered or not after calling each `Register` method.

```go
allJobRegistered := false
regCh := make(chan bool)
go func() {
    tracker.Register(q1)
    regCh <- false
    tracker.Register(q2)
    regCh <- true
}

allJobRegistered = <-regCh

(snip)
```

Maybe `time.Sleep` is not required for waiting `Register` method call.

Reply:
This way is always passing test, never reached `t.Error("2 jobs inserted")` and `t.Error("second job not registered")`.

I think that current solution is extend `time.Sleep` timeout.